### PR TITLE
Use prettier for code formatting

### PR DESCRIPTION
**Pull request by JuliaGerasymenko**

### CODE REVIEW

File: doc/doc.md [hunk 1]
```diff
@@ -35,44 +35,39 @@ Options:
...
 _Returns:_ [`<Object>`] function signature
+
 - `title`: [`<string>`] short function description
 - `description`: [`<string>`] extended function description
-- `parameters`: [`<Object[]>`][`<Object>`] function parameters, `{ name, types,
-      nonStandardTypes, comment, offset }`
-- `comments`: [`<Object[]>`][`<Object>`] comments about returned value, thrown
-      errors, deprecation and usage, `{ name, types, nonStandardTypes, comment
-      }`
+- `parameters`: [`<Object[]>`][`<object>`] function parameters, `{ name, types, nonStandardTypes, comment, offset }`
```
Comment (SemenchenkoVitaliy) [inline doc/doc.md]: `metadoc` moves text to the next line if it is longer than 80 characters. This is expected behaviour and I see no reason to change that.

File: doc/doc.md [hunk 2]
```diff
@@ -35,44 +35,39 @@ Options:
...
-    comments }`
+_Returns:_ [`<Map>`] hash of hash of records, `{ title, description, parameters, comments }`
 
 Introspect interface
 
-
 #### parseSignature(fn, text, start)
 
-- `fn`: [`<Function>`]` | `[`<RegExp>`] to be searched
+- `fn`: [`<Function>`]`|`[`<RegExp>`] to be searched
```
Comment (SemenchenkoVitaliy) [inline doc/doc.md]: I'd prefer to leave spaces but I'm fine either way.

File: doc/doc.md [hunk 3]
```diff
@@ -81,11 +76,10 @@ See GitHub for a full [list of contributors](https://github.com/metarhia/metadoc
 
 Licesed under MIT license. Copyright (c) 2018 Metarhia contributors
 
-
-[`<Object>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
-[`<Function>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function
-[`<RegExp>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
-[`<Map>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
-[`<Array>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array
+[`<object>`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object
```
Comment (SemenchenkoVitaliy) [inline doc/doc.md]: IMO, it would be better to not convert types to lowercase.

File: test/example.md [hunk 4]
```diff
@@ -35,26 +36,27 @@ Multi-line expanded function description. Note that all lines should have
 length of 80 or less characters
 
 _Throws:_ [`<TypeError>`] conditions causing error. Empty lines between comments
-    after parameters are optional
+after parameters are optional
```
Comment (SemenchenkoVitaliy) [inline test/example.md]: I'd prefer to leave this padding.

File: test/example.md [hunk 5]
```diff
@@ -35,26 +36,27 @@ Multi-line expanded function description. Note that all lines should have
...
-    `newMethodName` instead.
+have reason for deprecation and method to use instead if any. e.g. Removed
+due to incompatibility with `moduleName` in version 2.0.0. Use
+`newMethodName` instead.
 
 _Example:_
+
 ```js
- methodName(1, '2', {}, false, ['3'], data, {}, fn);
+methodName(1, '2', {}, false, ['3'], data, {}, fn);
```
Comment (SemenchenkoVitaliy) [inline test/example.md]: It'll be better to leave original example code formatting.

File: test/example.md [hunk 6]
```diff
@@ -63,69 +65,58 @@ _Result:_
...
 ExampleClass description
 
-
 #### ExampleClass.method2(num)
 
 - `num`: [`<number>`]
 
 method2 description
 
-
```
Comment (SemenchenkoVitaliy) [inline test/example.md]: Two empty lines were supposed to separate methods though I don't have a strong opinion on this one.

File: doc/doc.md [hunk 7]
```diff
@@ -35,44 +35,39 @@ Options:
...
 - `title`: [`<string>`] short function description
 - `description`: [`<string>`] extended function description
-- `parameters`: [`<Object[]>`][`<Object>`] function parameters, `{ name, types,
-      nonStandardTypes, comment, offset }`
-- `comments`: [`<Object[]>`][`<Object>`] comments about returned value, thrown
-      errors, deprecation and usage, `{ name, types, nonStandardTypes, comment
-      }`
+- `parameters`: [`<Object[]>`][`<object>`] function parameters, `{ name, types, nonStandardTypes, comment, offset }`
+- `comments`: [`<Object[]>`][`<object>`] comments about returned value, thrown
+  errors, deprecation and usage, `{ name, types, nonStandardTypes, comment }`
```
Comment (SemenchenkoVitaliy) [inline doc/doc.md]: All lines except for the first one are supposed to have padding. Also, why prettier didn't merge this line with the previous one?

File: .eslintrc.yml [hunk 8]
```diff
@@ -1 +1,7 @@
-extends: 'metarhia'
+extends:
+  - metarhia
+  - plugin:prettier/recommended
+plugins:
+  - prettier
+rules:
+  prettier/prettier: error
```
[the remark this answers is not in the document]
Comment (belochub) [reply .eslintrc.yml]: @JuliaGerasymenko This should also be done in other repositories as well.

File: .prettierignore [hunk 9]
```diff
@@ -0,0 +1,2 @@
+coverage
+dist
```
[the remark this answers is not in the document]
Comment (belochub) [reply .prettierignore]: @lundibundi they are in `.eslintignore` already in this repository, though. I don't see a problem with specifying ignored dirs that are possibly unnecessary at most times but may be useful in general when someone uses specific tools.

**Comment by SemenchenkoVitaliy** [comment]
Since it takes such a long time and I don't see a way to configure prettier to at least leave 2 empty lines between functions and cot not merge lines longer than 80 characters, I propose to add `doc/` & `test/` folders to `.prettierignore` and also remove autogenerated comments from `README.md`. @belochub @lundibundi @nechaido WDYT?

**Comment by lundibundi** [comment]
I ok with adding *auto-generated* and examples files to ignore file.  
Though not the whole `test` folder, I think we can create another root-level `examples` folder and put `example.js` there.

**Comment by SemenchenkoVitaliy** [comment]
@lundibundi Agree, though I don't think we should put `example.js` in a separate folder. We can just have `test/*.md` at `.prettierignore`. Either that or put `example.md` also in `examples` folder. My point is that example doc files should stay near to the files they were generated from so it will be easy to look and compare source file and autogenerated documentation.

**Pull request merged**